fmriprep/workflows/confounds.py

```diff
@@ -243,3 +264,211 @@ def reverse_order(inlist):
+
+    # melodic node
+    melodic = pe.Node(nws.MELODICRPT(no_bet=True,
+                                     no_mm=True,
+                                     generate_report=True),
+                      name="melodic")
+
+    # ica_aroma node
+    ica_aroma = pe.Node(aroma.ICA_AROMA(denoise_type='no'),
+                        name='ica_aroma')
+
+    # extract the confound ICs from the results
+    ica_aroma_confound_extraction = pe.Node(utility.Function(input_names=['ica_out_dir',
+                                                             'ignore_aroma_err'],
+                                                             output_names=['aroma_confounds',
+                                                                           'aroma_noise_ics',
+                                                                           'melodic_mix'],
+                                                             function=get_ica_confounds),
+                                            name='ica_aroma_confound_extraction')
+    ica_aroma_confound_extraction.inputs.ignore_aroma_err = ignore_aroma_err
+
+    # connect the nodes
+    workflow.connect([
+        # Connect input nodes to complete smoothing
+        (inputnode, calc_median_val,
+            [('epi_mni', 'in_file'),
+             ('epi_mask_mni', 'mask_file')]),
+        (calc_median_val, brightness_threshold,
+            [('out_stat', 'medianval')]),
+        (inputnode, calc_epi_mean,
+            [('epi_mni', 'in_file')]),
+        (calc_epi_mean, getusans,
+            [('out_file', 'image')]),
+        (calc_median_val, getusans,
+            [('out_stat', 'thresh')]),
+        (inputnode, smooth,
+            [('epi_mni', 'in_file')]),
+        (getusans, smooth,
+            [('usans', 'usans')]),
+        (brightness_threshold, smooth,
+            [('thresh', 'brightness_threshold')]),
+        # connect smooth to melodic
+        (smooth, melodic,
+            [('smoothed_file', 'in_files')]),
+        (inputnode, melodic,
+            [('epi_mask_mni', 'report_mask')]),
+        (inputnode, melodic,
+            [('epi_mask_mni', 'mask')]),
+        # connect nodes to ICA-AROMA
+        (smooth, ica_aroma,
+            [('smoothed_file', 'in_file')]),
+        (inputnode, ica_aroma,
+            [('movpar_file', 'motion_parameters')]),
+        (melodic, ica_aroma,
+            [('out_dir', 'melodic_dir')]),
+        # geneerate tsvs from ICA_AROMA
+        (ica_aroma, ica_aroma_confound_extraction,
+            [('out_dir', 'ica_out_dir')]),
+        # output for processing and reporting
+        (ica_aroma_confound_extraction, outputnode,
+            [('aroma_confounds', 'aroma_confounds'),
+             ('aroma_noise_ics', 'aroma_noise_ics'),
+             ('melodic_mix', 'melodic_mix')]),
+        # TODO change melodic report to reflect noise and non-noise components
+        (melodic, outputnode,
+            [('out_report', 'out_report')]),
```

Comment:
A lot of these connections can be reduced to one or two lines. I find this extremely hard to read as is.